src/Transport/Topology/Topologies/ForwardingTopology.cs

```diff
@@ -109,6 +110,14 @@ public Func<IDispatchMessages> GetDispatcherFactory()
             return () => container.Resolve<IDispatchMessages>();
         }
 
+        public StartupCheckResult ApplyPreStartupChecks()
+        {
+            var checks = container.ResolveAll<IStartupCheck>();
+            var compositeStartupChecker = new CompositeStartupCheck(checks);
```

Comment:
Something here feels odd @melkio. We register all the checks in `ApplyDefaults` to resolve those in `ApplyPreStartupChecks` so that we can construct `CompositeStartupCheck`?